### PR TITLE
Deal only with Compressed files

### PR DESCRIPTION
**Pull request by zaibon**

### CODE REVIEW

File: cache/fscache.go [hunk 1]
```diff
@@ -90,17 +90,12 @@ func (f *fsCache) SetMetaData([]string) error {
 func (f *fsCache) Open(path string) (io.ReadSeeker, error) {
 	chrootPath := chroot(f.root, filepath.Join(f.dedupe, path+".bro"))
 
-	var r io.ReadSeeker
-	var err error
-	r, err = os.Open(chrootPath) //try compressed file
+	r, err := os.Open(chrootPath) //try compressed file
 	if err == nil {
```
Comment (muhamadazmy) [inline cache/fscache.go]: Shouldn't this be if err != nil now. so you open the uncompressed file.

File: cache/fscache.go [hunk 2]
```diff
@@ -90,17 +90,12 @@ func (f *fsCache) SetMetaData([]string) error {
...
-		r = utils.NewReadSeeker(brotli.NewReader(r))
-	} else {
 		log.Debugf("Compressed file not available, try plain file. (%s)", chrootPath)
-		r, err = os.Open(chrootPath[:len(chrootPath)-4])
+		return nil, err
 	}
-
-	return r, err
+	return utils.NewReadSeeker(brotli.NewReader(r)), nil
```
Comment (muhamadazmy) [inline cache/fscache.go]: Does this mean only compressed files are supported now ?

Comment (muhamadazmy) [inline cache/fscache.go]: In that cacse, the ReadSeeker wrapper should be refactored for performance, currently it reads a head until the end of the file, may be it's better if we just read until the seek position. 

Also we might look at the OpenNonSeekable option, may be we can avoid using the NewReadSeeker all together.

**Comment by muhamadazmy** [comment]
My only concern is how much this will impact performance.

File: cache/fscache.go (hunk 2 above)
Comment (zaibon) [inline cache/fscache.go]: Well I think dropping completely uncompressed file is not a good idea actually. In jumpscale now we only build compressed file. But keeping both in this project make more sense. My bad updated this too quickly.

Comment (despiegk) [inline cache/fscache.go]: think should go completely compressed
why have both?

On Thu, Dec 10, 2015 at 1:34 PM, Christophe de Carvalho <
notifications@github.com> wrote:

> In cache/fscache.go
> <https://github.com/Jumpscale/aysfs/pull/8#discussion_r47220663>:
>
> >  	}
> > -
> > -	return r, err
> > +	return utils.NewReadSeeker(brotli.NewReader(r)), nil
>
> Well I think dropping completely uncompressed file is not a good idea
> actually. In jumpscale now we only build compressed file. But keeping both
> in this project make more sense. My bad updated this too quickly.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Jumpscale/aysfs/pull/8/files#r47220663>.
>



-- 
------------------------
Kristof De Spiegeleer
+971525609014 (my telegram account registered on this nr, best way to reach
me)
+201206927877
+32 475405474
skype: despiegk

**Comment by zaibon** [comment]
deprecated

**Pull request closed without merging**